81-IDevID_and_IAK_support.md

```diff
@@ -0,0 +1,287 @@
+ * mTLS usage scenario 
+ 
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+We propose to optionally leverage the IDevID and IAK keys/certificates to register each device at the registrar service and generate the LDevID and LAK keys/certificates. This will enable: 1) further check of the device based on IDevID information and CA trustchain ; 2) use of IAK for generation of LAK and LDevID ; 3) attestation based on LAK instead of an ephemeral AK.
+
+For compatibility reasons and taking into account different scenarios, the proposal extends to the ones detailed below regarding the possible keys. When EAK is mentioned it means an ephemeral Attestation Key that can be generated at the TPM with access to exporting it's public and private parts (not a Primary Key).
+ * EK, EK certificate and EAK: this is the current configuration used in Keylime
+ * EK, EK certificate, IDevID, IAK and EAK: this configuration can be used when an IDevID and IAK exists, but the user does not want to maintain a CA for issuing LDevIDs or LAKs.
+ * EK, EK certificate, LDevID and LAK: this configuration can be used when a persistent AK is required but no IDevID is available, linking the LDevID on the EK.
+ * EK, EK certificate, IDevID, IAK, LDevID and LAK: this configuration enables issuing LDevID and LAKs based on the IAK (from the OEM) and allows the user/client to issue LAK certificates from this own CA.
+
+Adding support for IDevID and IAK as an option to the Keylime registrar and verifier services, to the RUST agent, will allow our users to take advantage of IDevID and IAK when using Keylime. It will promote use of IDevID for switches and servers, which will improve security for all users.
+Using the IAK and IDevID credentials would mean the OEM had already exercised the proof of residency pre-requisites for generating the credentials, making it possible to simplify the registering process to a single exchange from the agent to the registrar service, by skipping the DevID provisioning in the field.
+
+If required, it would be possible to send a challenge (nonce) to a node agent to prove it owns the TPM protected private IAK key. A nonce challenge to make sure that the node can access the IAK private key is not mandatory but would be good just for avoiding registering a node that is invalid (does not have access to the private key related to the certificate provided). Even without this, the node would not be able to attest later.
+
+Finally, further planning will be required on how the IAK certificates will be accepted. The registrar service will need to be configured with a set of trusted CA certificates for validating IAK certificates in order to support a multi-OEM environments.
```

Comment:
I think this proposal should include how the validation is done.

Reply:
Agreed, still working on this point, will update the PR soon